fastavro/_schema_py.py

```diff
@@ -267,16 +267,36 @@ def _parse_schema(
         logical_type = parsed_schema.get("logicalType")
         if logical_type == "decimal":
             scale = parsed_schema.get("scale")
-            if scale and not isinstance(scale, int):
+            if scale and (not isinstance(scale, int) or scale < 0):
                 raise SchemaParseException(
                     "decimal scale must be a postive integer, "
                     + "not {}".format(scale)
                 )
+
             precision = parsed_schema.get("precision")
-            if precision and not isinstance(precision, int):
+            if precision:
+                if not isinstance(precision, int) or precision <= 0:
+                    raise SchemaParseException(
+                        "decimal precision must be a postive integer, "
+                        + "not {}".format(precision)
+                    )
+                if schema_type == "fixed":
+                    # https://avro.apache.org/docs/current/spec.html#Decimal
+                    size = schema["size"]
+                    max_precision = int(
+                        math.floor(math.log10(2) * (8 * size - 1))
```

Comment:
I see that this is the same formula that used in the apache avro libraries, but isn't it ever so slightly different than the formula on the spec page? The spec page has an extra `- 1` near the end. In reality it doesn't seem to matter because we are doing a floor function and the `size` can only be whole integers. I'm guessing the final `- 1` is dropped to simplify the function.

Or am I misreading the function in the spec?

Reply:
Yes, you are right. The -1 part is dropped because in the correct full formula: log10(2^(8*n-1)) + log10(1-1/2^(8*n-1)), the last part will go to 0 with bigger 'n'.

Reply:
And, btw, I took this simplified formula from reference official avro C++ library.